Lib/ntpath.py

```diff
@@ -554,36 +554,46 @@ def normpath(path):
         return prefix + sep.join(comps)
 
 
-def _abspath_fallback(path):
-    """Return the absolute version of a path as a fallback function in case
-    `nt._getfullpathname` is not available or raises OSError. See bpo-31047 for
-    more.
-
-    """
-
-    path = os.fspath(path)
-    if not isabs(path):
-        if isinstance(path, bytes):
-            cwd = os.getcwdb()
-        else:
-            cwd = os.getcwd()
-        path = join(cwd, path)
-    return normpath(path)
-
 # Return an absolute path.
 try:
-    from nt import _getfullpathname
+    from nt import _path_abspath
 
 except ImportError: # not running on Windows - mock up something sensible
-    abspath = _abspath_fallback
+    def abspath(path):
+        """Return the absolute version of a path."""
+        path = os.fspath(path)
+        if not isabs(path):
+            if isinstance(path, bytes):
+                path = join(os.getcwdb(), path)
+            else:
+                path = join(os.getcwd(), path)
```

Comment:
This fallback implementation is wrong for drive-relative paths. `abspath()` shouldn't return a relative path such as "Z:spam". It needs to assume that the working directory on each drive is the root directory. Frankly, joining a Windows path with the working directory on POSIX is just gobbledygook nonsense.